variables.tf

```diff
@@ -393,6 +403,14 @@ variable "existing_subnets" {
   }))
   default  = []
   nullable = false
+
+  validation {
+    condition = (
+      (var.create_subnets && length(var.existing_subnets) == 0) ||
+      (!var.create_subnets && length(var.existing_subnets) > 0)
+    )
+    error_message = "You must either set 'create_subnets' to true and not provide 'existing_subnets', or set it to false and provide a non-empty list of 'existing_subnets'."
```

Comment:
error message can be changed to:
`You must either set 'create_subnets' to true and leave 'existing_subnets' empty, or set 'create_subnets' to false and provide a non-empty list for 'existing_subnets'.`